### PR TITLE
feat: init containers

### LINKED ISSUE #30: Missing initContainer in PodSpecProps
Kubernetes has it https://kubernetes.io/docs/concepts/workloads/pods/init-containers/ but not cdk8s

When try to use it

image

```Typescript
/**
 * (experimental) Properties of a `PodSpec`.
 *
 * @experimental
 */
export interface PodSpecProps {
    /**
     * (experimental) List of containers belonging to the pod.
     *
     * Containers cannot currently be
     * added or removed. There must be at least one container in a Pod.
     *
     * You can add additionnal containers using `podSpec.addContainer()`
     *
     * @default - No containers. Note that a pod spec must include at least one container.
     * @experimental
     */
    readonly containers?: ContainerProps[];
    /**
     * (experimental) List of volumes that can be mounted by containers belonging to the pod.
     *
     * You can also add volumes later using `podSpec.addVolume()`
     *
     * @default - No volumes.
     * @see https://kubernetes.io/docs/concepts/storage/volumes
     * @experimental
     */
    readonly volumes?: Volume[];
    /**
     * (experimental) Restart policy for all containers within the pod.
     *
     * @default RestartPolicy.ALWAYS
     * @see https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy
     * @experimental
     */
    readonly restartPolicy?: RestartPolicy;
    /**
     * (experimental) A service account provides an identity for processes that run in a Pod.
     *
     * When you (a human) access the cluster (for example, using kubectl), you are
     * authenticated by the apiserver as a particular User Account (currently this
     * is usually admin, unless your cluster administrator has customized your
     * cluster). Processes in containers inside pods can also contact the
     * apiserver. When they do, they are authenticated as a particular Service
     * Account (for example, default).
     *
     * @default - No service account.
     * @see https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
     * @experimental
     */
    readonly serviceAccount?: IServiceAccount;
}
```

### PR DESCRIPTION
**Pull request by iliapolo**
Fixes https://github.com/cdk8s-team/cdk8s-plus/issues/30

Signed-off-by: iliapolo <epolon@amazon.com>

### CODE REVIEW

File: src/pod.ts [hunk 1]
```diff
@@ -105,6 +128,30 @@ export class PodSpec implements IPodSpec {
...
+      throw new Error('Init containers must not have a liveness probe');
+    }
+
+    if (container.startup) {
+      throw new Error('Init containers must not have a startup probe');
+    }
+
+    const impl = new Container({
+      ...container,
+      name: container.name ?? `init-${this._initContainers.length}`,
```
Comment (Chriscbr) [inline src/pod.ts:148]: Clever :)

File: test/pod.test.ts [hunk 2]
```diff
@@ -194,3 +195,98 @@ test('Synthesizes spec lazily', () => {
...
+test('init containers cannot have startup probe', () => {
+
+  const chart = Testing.chart();
+  const pod = new kplus.Pod(chart, 'Pod', { containers: [{ image: 'image' }] });
+
+  expect(() => pod.addInitContainer({ image: 'image', startup: Probe.fromTcpSocket() })).toThrow('Init containers must not have a startup probe');
+
+});
+
+test('can specfity init containers at instatiation', () => {
```
Comment (Chriscbr) [inline test/pod.test.ts]:
```suggestion
test('can specify init containers at instantiation', () => {
```

**Review by Chriscbr — APPROVED** [review]
Nice work 👍

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-21/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/560)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-20/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/564)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**